src/compile/data/transforms.ts

```diff
@@ -65,6 +66,9 @@ export function parseTransformArray(model: Model) {
       node = new CalculateNode(t);
     } else if (isFilter(t)) {
       node = new FilterNode(model, t.filter);
+    } else if (isBin(t)) {
+      const modelWithField = model as ModelWithField;
+      node = BinNode.makeTransform(modelWithField, t);
```

Comment:
1) As mentioned a few times earlier, in general, you should never cast anything in the main code without a good reason.

2) In this case, I don't see why the model here should be a `ModelWithField` -- it should just be `Model`.